File: src/test/java/uk/gov/justice/hmpps/offenderevents/services/wiremock/PrisonApiExtension.java
package uk.gov.justice.hmpps.offenderevents.services.wiremock;

import org.junit.jupiter.api.extension.AfterAllCallback;
import org.junit.jupiter.api.extension.BeforeAllCallback;
import org.junit.jupiter.api.extension.BeforeEachCallback;
import org.junit.jupiter.api.extension.ExtensionContext;

public class PrisonApiExtension implements BeforeAllCallback, AfterAllCallback, BeforeEachCallback {
    public static final PrisonApiMockServer server = new PrisonApiMockServer();

    @Override
    public void afterAll(ExtensionContext context) {
        server.stop();
    }

    @Override
    public void beforeAll(ExtensionContext context) {
        server.start();
    }

    @Override
    public void beforeEach(ExtensionContext context) {
        server.resetRequests();
    }
}

File: src/test/kotlin/uk/gov/justice/hmpps/offenderevents/services/wiremock/PrisonApiExtension.kt
package uk.gov.justice.hmpps.offenderevents.services.wiremock

import org.junit.jupiter.api.extension.AfterAllCallback
import org.junit.jupiter.api.extension.BeforeAllCallback
import org.junit.jupiter.api.extension.BeforeEachCallback
import org.junit.jupiter.api.extension.ExtensionContext

class PrisonApiExtension :
  BeforeAllCallback,
  AfterAllCallback,
  BeforeEachCallback {
  override fun afterAll(context: ExtensionContext): Unit = server.stop()
  override fun beforeAll(context: ExtensionContext): Unit = server.start()
  override fun beforeEach(context: ExtensionContext): Unit = server.resetRequests()

  companion object {
    val server = PrisonApiMockServer()
  }
}
